plugins/woocommerce-blocks/assets/js/blocks/product-collection/types.ts

```diff
@@ -31,6 +41,12 @@ export interface ProductCollectionAttributes {
 	filterable: boolean;
 	// eslint-disable-next-line @typescript-eslint/naming-convention
 	__privatePreviewState?: PreviewState;
+	selectedReference?: ProductCollectionSelectedReference;
+}
+
+export interface ProductCollectionSelectedReference {
+	type: 'product'; // In future, we can add more types like 'category', 'tag' etc.
```

Comment:
AFAIU It will only contain selected reference, which means in case of `product` reference we will have specific product id. I don't think we have near term plan to allow users to select specific `order` or taxonomy. But if we will allow that in future, then I believe we will still have a specific id for that.

AFAIU `WooCommerceBlockLocation` is used to determine where the block is being used. But `selectedReference` is used when a reference needs to be selected manually by merchant. From this POV, those are different things. 

> Ideally this looks like it can be replaced by WooCommerceBlockLocation

While thinking about this, I have improved the type of `WooCommerceBlockLocation` to be more specific in 53e1da8. Does that look good to you? 🙂

After making above change, I think we can use `WooCommerceBlockLocation` type for `selectedReference` too. This way, we are storing both of these in same format. Although, I am not 100% sure about this. Because as I explained above, we won't need anything other than specific id for `selectedReference`. But I am open to suggestions. 

Although both may look similar, I am not sure if they are the same. You may think that it does not follow the DRY principle, but the intent here is different, so we are not violating the DRY principle.

CC: @Kmanijak What do you think about this? 🤔

Reply:
> Although, I am not 100% sure about this. Because as I explained above, we won't need anything other than specific id for selectedReference. But I am open to suggestions.

@imanish003 Yeah, I've been thinking about this as well. Theoretically, having just one `productReference: number` attribute should suffice. 

I understand that using an object with `type` is meant to accommodate additional manual types in the future, but let’s consider a scenario where we, indeed, want to support the selection of manual products **and** terms and:
- A collection with `usesReference: [product, archive]`
- In that case, we would need to store information about **both**. This could simply be resolved by using separate attributes _(e.g., productReference, termReference, etc.)._

Based on the example above, the current structure would need to become an array of _ProductCollectionSelectedReference's_ to support multiple references. Instead, having dedicated attributes has some pros:
- It's simple to start with; especially since we are not sure of supporting any other type, and;
- It is easier to freehand write PC blocks.

What do you folks think? Should we simplify to storing a single number here?

Reply:
I don't have a strong opinion here. I am fine with storing a single number. Later, if needed, we can use the `deprecated` API to update the structure.

> **Update:** If that's something not worth pursuing, we can agree that this will require an array of 'WooCommerceBlockLocation's as the selected reference if more manual types are needed and move on. Totally fine with that.

In case we decide to go with the current structure, we don't need to make it an array now. Later, we can use a `deprecated` API to update the structure. In future, it will look like:
```
[
  {
    type: 'product',
    id: 123
  },
  {
    type: 'archive',
    id: 456
  }
]
```

I am more inclined towards this because we can keep all selected references in a single attribute, which will reduce developers' cognitive load. What do you think?

Reply:
As we discussed in sync call. I have renamed `selectedReference` to `productReference`. Also, `productReference` will be number instead of object.